[Andy Lam] iP

lamwj98 opened the pull request:
# Siri
Siri helps you keep track of the task you have on hand so that you can stop thinking about what you have to complete and focus on completing what you need to complete!

> “If you spend too much time thinking about a thing, you’ll never get it done.” – Bruce Lee [(source)](https://dansilvestre.com/productivity-quotes/)

Its's,

- text-based 
- **EASY** and **SIMPLE** to use 😊
- **FREE**

Current version allows you to:

- [ ] Add/Remove tasks
- [ ] Mark/Unmark tasks
- [ ] List tasks
- [ ] Find tasks from the list

There are **three** _types_ of tasks available for use:

1. _deadline_ (tasks with a due date and an optional due time) e.g. `deadline something /by 12-12-2021`
2. _event_ (tasks with date and time) e.g. `event something /at 12-12-2021 15:00`
3. _todo_ (tasks without any date and time field) e.g. `todo something`


---
Here's a piece of code from Siri `main` class. You can change the path of storage by changing the Siri constructor's string to you file path.
```
public class Siri {
      public static void main(String[] args) {
          new Siri("../data/data.txt").runApp();
      }
}
```

Commit:
Commit for level-1

Commit:
Level1 v2

Commit:
level-2 completed

Commit:
Level-3 completed

Commit:
completed level-4

Commit:
Testing added

Commit:
Level-5 without additional test cases

Commit:
Level-5 completed with testing

Commit:
level-6 with test cases updated

Commit:
Merge with tmp2 - detached head Level-6

Commit:
Merge with tmp - detached head A-TextUiTesting

Commit:
Complete Level-7 without Merge

Commit:
Complete Level-8 without merge

Commit:
Merge branch 'branch-Level-8'

Commit:
Complete more oop without merging

Commit:
Complete creating package

Commit:
Complete A-JUnit

zihaowrez commented on src/main/java/siri/Deadline.java:
```diff
@@ -0,0 +1,80 @@
+
+    private static DateTimeFormatter dtf = DateTimeFormatter.ofPattern("dd-LLL-yyyy");
+    private LocalDate dlDate;
+    private LocalTime dlTime;
```
Perhaps it's better to use full words for the variable names (e.g. `dueLocalDate` and `dueLocalTime`)?

Similar for `dlLst`, `eLst` etc

zihaowrez commented on src/main/java/siri/Deadline.java:
```diff
@@ -0,0 +1,80 @@
+        this.dlTime = dlTime;
+    }
+
+    public boolean dateCompare(LocalDate date) {
```
Should the method be named `compareDate`?

Commit:
Completed A-JavaDoc without merge

zihaowrez commented on src/main/java/siri/Parser.java:
```diff
@@ -0,0 +1,205 @@
+
+    //deals with making sense of the user command
+    
+    public int handleCommand(String s) throws SiriException{
```
This is a long method. Is it better to shorten it using abstraction or other methods?

zihaowrez commented on src/main/java/siri/Storage.java:
```diff
@@ -0,0 +1,83 @@
+
+class Storage {
+    //deals with loading tasks from the file and saving tasks in the file
+    private File dirFile;
```
Maybe just name it as `directory`?

zihaowrez commented on src/main/java/siri/Task.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    /*
+        Method to return a symbol the doneness of the task:
```
Seems that the Javadoc's format is incorrect?

zihaowrez commented on src/main/java/siri/Task.java:
```diff
@@ -0,0 +1,62 @@
+ */
+
+class Task {
+    protected int done;
```
Consider using a boolean `isDone`?

zihaowrez commented on src/main/java/siri/TaskList.java:
```diff
@@ -0,0 +1,200 @@
+        lst.get(index).markDone();
+    }
+
+    public void printEventOn(LocalDate date) {
```
Good naming of the method!

zihaowrez reviewed (commented):
Good class names and method names. For variables, it might be better to avoid abbreviations.

Commit:
Complete A-CodingStandard without merge

Commit:
Complete A-CodingStandard with merge

Commit:
Completed Level-9 without merge

Commit:
Complete Level-9 with merge

Commit:
Completed A-Gradle

Commit:
Complete CheckStyle

Commit:
Complete GUI

j4ck990 commented on src/main/java/siri/DialogBox.java:
```diff
@@ -0,0 +1,62 @@
+    }
+
+    public static DialogBox getSiriDialog(String text, Image img) {
+        var db = new DialogBox(text, img);
```
Perhaps dialogBox can be spelled in full?
I have noticed other people using the name db for database as well.

j4ck990 commented on src/main/java/siri/Storage.java:
```diff
@@ -0,0 +1,111 @@
+    public Storage(String filePath) {
+
+        if (filePath == "") {
+            this.filePath = "data/data.txt";
```
Perhaps you can consider using various path separator variables from File, Files, or use Path imports? 
Helps account for different OS path separators. :)

j4ck990 commented on src/main/java/siri/Parser.java:
```diff
@@ -0,0 +1,226 @@
+     * @throws SiriException if command structure is invalid, command is not recognised or
+     * if there are missing or extra fields in command.
+     */
+    public String handleCommand(String command) throws SiriException {
```
Perhaps extracting out logic for each case block would help with the long method?

j4ck990 commented on src/main/java/siri/Task.java:
```diff
@@ -0,0 +1,72 @@
+     * @param done integer to indicate whether done or undone. 1 represents the done and 0 represents undone.
+     * @return a string " " to indicate the task had not been completed and "X" to indicate the task had be completed.
+     */
+    private String doneSymbol(boolean isDone) {
```
Perhaps this method can be renamed into something that hints at more of a getter? e.g. getDoneSymbol()

Commit:
Add assertion to functinal code in Storage and TaskList classes

In TaskList class, assertions had been made to the following methods:

  - addItem
  - deleteTask
  - markItem
  - unmarkItem

In Storage class, assertion had been made to the follow method:

  - load

These assertions are made to ensure that the methods performs the required
function that it is expected to do.

In addition, minor modifications had been made in the build.gradle file to
ensure that the jar file built is functional. These changes include modifying
the mainClassName of application to "siri.Launcher" as well as archiveBaseName
to "siri".

Commit:
Modify code to increase code quality

Modified the following classes:
  - Parser
  - Storage
  - TaskList

In Parser class, handleCommand function had been modified to increase code
quality as it had been previously lump with a single large switch case
making code readbility low. The following private functions had been
added to extract out the individual commands to be processed separately:

  - handleMarkCommand
  - handleUnmarkCommand
  - handleListCommand
  - handleTodoCommand
  - handleDeadlineCommand
  - handleEventComamand
  - handleDeleteCommand
  - handleEPrintCommand
  - handleDlPrintCommand
  - handleFindCommand

In TaskList class, JavaDocs had been modified for the following classes to
reflect the return value of the methods:

  - addItem
  - deleteItem
  - print
  - markItem
  - unmarkItem
  - printEventOn
  - printDeadlineOn
  - find

Additionally, changes had been made to addItem, deleteItem, markItem and
unmarkItem functions to process and throw list related exceptions such as
exception relating to list length as well as item index of list. A new
private function had been added to create a Deadline object base on the
loaded data so that other functions relying to create a new Deadline
object do not need to check whether time field is present. The error in
code for find had been fixed to from printing on console to returning
a string when tmp size is 0.

In Storage, load function had been modified to reduce the number of
indentation and make the code clearer.

lamwj98 committed:
Merge pull request #1 from lamwj98/branch-A-Assertions

Add assertions to functional code

Commit:
Merge branch 'branch-A-CodeQuality'

Modify code to increase code quality

Modified the following classes:

  - Parser
  - Storage
  - TaskList

In Parser class, handleCommand function had been modified to increase code
quality as it had been previously lump with a single large switch case
making code readbility low. The following private functions had been
added to extract out the individual commands to be processed separately:

  - handleMarkCommand
  - handleUnmarkCommand
  - handleListCommand
  - handleTodoCommand
  - handleDeadlineCommand
  - handleEventComamand
  - handleDeleteCommand
  - handleEPrintCommand
  - handleDlPrintCommand
  - handleFindCommand

In TaskList class, JavaDocs had been modified for the following classes to
reflect the return value of the methods:

  - addItem
  - deleteItem
  - print
  - markItem
  - unmarkItem
  - printEventOn
  - printDeadlineOn
  - find

Additionally, changes had been made to addItem, deleteItem, markItem and
unmarkItem functions to process and throw list related exceptions such as
exception relating to list length as well as item index of list. A new
private function had been added to create a Deadline object base on the
loaded data so that other functions relying to create a new Deadline
object do not need to check whether time field is present. The error in
code for find had been fixed to from printing on console to returning
a string when tmp size is 0.

In Storage, load function had been modified to reduce the number of
indentation and make the code clearer.

Commit:
Complete C-MassOps

Commit:
Completed B-DetectAnomalies

Commit:
Completed B-DetectAnomalies

Add new extensions

Added the following extensions to the project:

  - Detect anomalies when task is being added (clashes between tasks)
  - Mass operations allowed for delete, mark and unmark

In Event class, new getters had been set to get the date and time for the
event item.

In Parser class, a private function had been create to generate a list of
interger from a string passed.

In TaskList class, deleteItem, markItem and unmarkItem had been modified to allow
manipulation of multiple task at one time. In addItem, error will be thrown if
user want to add a new event and the date and time of event clashes with existing
event.

lamwj98 committed:
Create workflows

lamwj98 committed:
Delete workflows

Commit:
Clean up code withr checkstyle

Commit:
Clean up code with checkstyle

Commit:
Fix issues by nus-se-bot and test bugs

Commit:
Clean up more checkstyle errors

lamwj98 committed:
Create gradle.yml

lamwj98 committed:
Merge pull request #6 from lamwj98/lamwj98-patch-1-1

Create gradle.yml

Commit:
Modify code to have better GUI

Modifications made in current commit:
- Allow text printed to be wrapped and not truncated
- Print error messages in red.
- Allow start up message to be printed (with the processed first command)
- When user types "bye", app saves data and close JavaFX app

Commit:
Complete User Guide

Provided a user guide - docs/READ.md which consist of the following details:
- Description of the application
- Where to download the application
- Commands in the application
- Acknowledgements of resources used

Commit:
Fix docs/READ.md file

Commit:
Change READ.md file

lamwj98 committed:
Update READ.md

Commit:
Fix file path issue in jar

Commit:
Merge with upstream

lamwj98 committed:
Delete data directory

Directory is per run data and should not be here.

lamwj98 committed:
Delete src/main/data directory

Directory is per run data and should not be here.

lamwj98 committed:
Delete README.md

lamwj98 committed:
Update Ui.java

lamwj98 committed:
Update DialogBox.java

lamwj98 committed:
Update MainWindow.java
